opus/import/populate_obs_instrument_VGUVS_prof.py

```diff
@@ -0,0 +1,611 @@
+
+    return ea
+
+# We set the center versions to be the same as the normal versions
+populate_obs_ring_geometry_VGUVS_center_phase1_PROF = \
+    populate_obs_ring_geometry_VGUVS_phase1_PROF
+populate_obs_ring_geometry_VGUVS_center_phase2_PROF = \
+    populate_obs_ring_geometry_VGUVS_phase2_PROF
+populate_obs_ring_geometry_VGUVS_center_incidence1_PROF = \
+    populate_obs_ring_geometry_VGUVS_incidence1_PROF
+populate_obs_ring_geometry_VGUVS_center_incidence2_PROF = \
+    populate_obs_ring_geometry_VGUVS_incidence2_PROF
+populate_obs_ring_geometry_VGUVS_center_emission1_PROF = \
+    populate_obs_ring_geometry_VGUVS_emission1_PROF
+populate_obs_ring_geometry_VGUVS_center_emission2_PROF = \
+    populate_obs_ring_geometry_VGUVS_emission2_PROF
+populate_obs_ring_geometry_VGUVS_center_north_based_incidence1_PROF = \
+    populate_obs_ring_geometry_VGUVS_north_based_incidence1_PROF
+populate_obs_ring_geometry_VGUVS_center_north_based_incidence2_PROF = \
+    populate_obs_ring_geometry_VGUVS_north_based_incidence2_PROF
+populate_obs_ring_geometry_VGUVS_center_north_based_emission1_PROF = \
+    populate_obs_ring_geometry_VGUVS_north_based_emission1_PROF
+populate_obs_ring_geometry_VGUVS_center_north_based_emission2_PROF = \
+    populate_obs_ring_geometry_VGUVS_north_based_emission2_PROF
+
+
+# Opening angle to observer: the angle between the ring surface to the direction
+# where outgoing photons to the observer. Positive if observer is at the north
+# side of the ring , negative if it's at the south side. If observer is at the
+# north side, it's ea - 90 (or 90 - inc). If observer is at the south side,
+# then oa is 90 - ea.
+def populate_obs_ring_geometry_VGUVS_observer_ring_opening_angle1_PROF(**kwargs):
+    if _is_Voyager_at_north(**kwargs):
+        min_ea = populate_obs_ring_geometry_VGUVS_emission1_PROF(**kwargs)
+        return min_ea - 90.
+    else:
+        max_ea = populate_obs_ring_geometry_VGUVS_emission2_PROF(**kwargs)
+        return 90. - max_ea
+
+def populate_obs_ring_geometry_VGUVS_observer_ring_opening_angle2_PROF(**kwargs):
+    if _is_Voyager_at_north(**kwargs):
+        max_ea = populate_obs_ring_geometry_VGUVS_emission2_PROF(**kwargs)
+        return max_ea - 90.
+    else:
+        min_ea = populate_obs_ring_geometry_VGUVS_emission1_PROF(**kwargs)
+        return 90. - min_ea
+
+# Ring elevation to observer, same to opening angle except, it's positive if
+# observer is at north side of Jupiter, Saturn, and Neptune, and south side of
+# Uranus. Negative if observer is at south side of Jupiter, Saturn, and Neptune,
+# and north side of Uranus.
+def populate_obs_ring_geometry_VGUVS_observer_ring_elevation1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+    if _is_Voyager_at_north(**kwargs):
+        target_name = populate_target_name_from_index(**kwargs)
```

Comment:
How about a new function called something like _is_voyager_at_north_except_uranus. Then you don't have to hard-code the target name stuff in every function here and you can remove one level of IFs as well.